framework/src/play/libs/ws/WSAsync.java

```diff
@@ -663,6 +667,43 @@ public InputStream getStream() {
             }
         }
 
+        @Override
+        public Document getXml(String encoding) {
+            try {
+                checkConsumed();
+                return super.getXml(encoding);
+            } finally {
+                consumed = true;
```

Comment:
are you sure every time `super.getXml()` throws an exception we should mark it as `consumed`?